src/Veldrid/D3D11/D3D11TextureView.cs

```diff
@@ -30,58 +31,47 @@ public D3D11TextureView(D3D11GraphicsDevice gd, ref TextureViewDescription descr
-                            uavDesc.Texture2DArray.MipSlice = (int)description.BaseMipLevel;
-                            uavDesc.Texture2DArray.FirstArraySlice = (int)description.BaseArrayLayer;
-                            uavDesc.Texture2DArray.ArraySize = (int)description.ArrayLayers;
-                        }
-                    }
-                }
-                else
-                {
-                    uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture3D;
-                    uavDesc.Texture3D.MipSlice = (int)description.BaseMipLevel;
+                    case TextureViewType.View1D:
+                        uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture1D;
+                        uavDesc.Texture1D.MipSlice = (int)description.BaseMipLevel;
+                        break;
+                    case TextureViewType.View1DArray:
+                        uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture1DArray;
+                        uavDesc.Texture1DArray.MipSlice = (int)description.BaseMipLevel;
+                        uavDesc.Texture1DArray.FirstArraySlice = (int)description.BaseArrayLayer;
+                        uavDesc.Texture1DArray.ArraySize = (int)description.ArrayLayers;
+                        break;
+
+                    case TextureViewType.View2D:
+                        uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture2D;
+                        uavDesc.Texture2D.MipSlice = (int)description.BaseMipLevel;
+                        break;
+                    case TextureViewType.View2DArray:
+                    case TextureViewType.ViewCube:
+                    case TextureViewType.ViewCubeArray:
+                        uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture2DArray;
+                        uavDesc.Texture2DArray.MipSlice = (int)description.BaseMipLevel;
+                        uavDesc.Texture2DArray.FirstArraySlice = (int)description.BaseArrayLayer;
+                        uavDesc.Texture2DArray.ArraySize = (int)description.ArrayLayers;
+                        break;
+
+                    case TextureViewType.View3D:
+                        uavDesc.ViewDimension = UnorderedAccessViewDimension.Texture3D;
+                        uavDesc.Texture3D.MipSlice = (int)description.BaseMipLevel;
 
-                    // Map the entire range of the 3D texture.
-                    uavDesc.Texture3D.FirstWSlice = 0;
-                    uavDesc.Texture3D.WSize = (int)d3dTex.Depth;
+                        // Map the entire range of the 3D texture.
+                        uavDesc.Texture3D.FirstWSlice = 0;
+                        uavDesc.Texture3D.WSize = (int)d3dTex.Depth;
+                        break;
                 }
 
                 UnorderedAccessView = device.CreateUnorderedAccessView(d3dTex.DeviceTexture, uavDesc);
             }
         }
 
-        public override string Name
+        public override string? Name
```

Comment:
This doesn't really seem to be related, although I am supportive of trying to correctly annotate nullability in the library over time. Could you remove it from this PR, though?

Reply:
Done